esmvalcore/preprocessor/_io.py

```diff
@@ -179,6 +179,41 @@ def concatenate(cubes):
     raise ValueError(f'Can not concatenate cubes: {msg}')
 
 
+def concatenate(cubes):
+    """Concatenate all cubes after fixing metadata."""
+    _fix_cube_attributes(cubes)
+
+    if len(cubes) > 2:
+        cubes_dict = {cube: cube.coord("time").points[0] for cube in cubes}
+        cubes_dict = OrderedDict(
```

Comment:
Note that all dicts are ordered since python 3.6, so there is no need to use OrderedDict anymore